qiskit_braket_provider/providers/braket_job.py

```diff
@@ -135,15 +136,31 @@ def cancel(self):
             task.cancel()
 
     def status(self):
-        braket_tasks_states = [task.state() for task in self._tasks]
+        braket_jobs_states = [task.state() for task in self._tasks]
 
-        if "FAILED" in braket_tasks_states:
+        if "FAILED" in braket_jobs_states:
             status = JobStatus.ERROR
-        elif "CANCELLED" in braket_tasks_states:
+        elif "CANCELLED" in braket_jobs_states:
             status = JobStatus.CANCELLED
-        elif all(state == "COMPLETED" for state in braket_tasks_states):
+        elif all(state == "COMPLETED" for state in braket_jobs_states):
             status = JobStatus.DONE
         else:
             status = JobStatus.RUNNING
 
         return status
+
+
+class AWSBraketJob(AmazonBraketTask):
+    def __init_subclass__(cls, **kwargs):
+        """This throws a deprecation warning on subclassing."""
+        warn(f"{cls.__name__} is deprecated.", DeprecationWarning, stacklevel=2)
+        super().__init_subclass__(**kwargs)
+
+    def __init__(self, *args, **kwargs):
```

Comment:
What happens if a user instantiates this class with `job_id` ? Could we please add a test to assert that there are no breaking changes with the introduction of this class?

Reply:
@kshitijc I have added a test for this, however for some reason, when I run it, there is a failed test for `TestAWSBraketJob`: `self.assertEqual(job.shots, 10)` (line 63)

The metadata are not passed for the old class. In principle, the old class (which inherits from the new class) should also be inheriting from JobV1. JobV1 allows us to set the metadata, which is not working correctly for `AWSBraketJob`. Any suggestion? It seems to be a bug.

you can see the code here: https://github.com/robotAstray/qiskit-braket-provider/blob/edaba8d167f8a734e845b6c774cae4174c9f2223/tests/providers/test_braket_job.py
While the error is this: 
```
FAIL: test_AWS_job (tests.providers.test_braket_job.TestAWSBraketJob)
Tests job. via AWSBraketJob
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/home/robotastray/Desktop/qiskit-braket-provider/tests/providers/test_braket_job.py", line 65, in test_AWS_job
    self.assertEqual(job.shots, 10)
AssertionError: 0 != 10

```

Reply:
This should be resolved in the latest commit and the code can be found here: https://github.com/robotAstray/qiskit-braket-provider/blob/5e05e45c39bc9582ecba68e399c4cfcb90119f83/tests/providers/test_braket_job.py